commands/azureCommands/acr-build-logs.ts

```diff
@@ -0,0 +1,381 @@
+import { BlobService, createBlobServiceWithSas } from "azure-storage";
+import * as path from 'path';
+import * as vscode from "vscode";
+import { AzureImageNode, AzureLoadingNode, AzureNotSignedInNode, AzureRegistryNode, AzureRepositoryNode } from '../../explorer/models/azureRegistryNodes';
+import ContainerRegistryManagementClient from "../../node_modules/azure-arm-containerregistry";
+import { Subscription } from "../../node_modules/azure-arm-resource/lib/subscription/models";
+import { getSubscriptionFromRegistry } from '../../utils/Azure/acrTools';
+import { AzureCredentialsManager } from '../../utils/azureCredentialsManager';
+import { quickPickACRRegistry } from '../utils/quick-pick-azure'
+
+const teleCmdId: string = 'vscode-docker.buildTaskLog';
+
+/**  This command is used through a right click on an azure registry, repository or image in the Docker Explorer. It is used to view build logs for a given item. */
+export async function viewBuildLogs(context: AzureRegistryNode | AzureRepositoryNode | AzureImageNode): Promise<void> {
+    let registry: Registry;
+    let subscription: Subscription;
+    if (!context) {
+        registry = await quickPickACRRegistry();
+        if (!registry) { return; }
+        subscription = getSubscriptionFromRegistry(registry);
+    } else {
+        registry = context.registry;
+        subscription = context.subscription;
+    }
+    let resourceGroup: string = registry.id.slice(registry.id.search('resourceGroups/') + 'resourceGroups/'.length, registry.id.search('/providers/'));
+
+    const client = AzureCredentialsManager.getInstance().getContainerRegistryManagementClient(subscription);
+    let logData: LogData = new LogData(client, registry, resourceGroup);
+    const filterFunction = context ? getFilterFunction(context) : undefined;
+    try {
+        await logData.loadMoreLogs(filterFunction);
+    } catch (error) {
+        if (error.code !== "NoRegisteredProviderFound") {
+            throw error;
+        }
+    }
+
+    if (logData.logs.length === 0) {
+        let itemType: string;
+        if (context && context instanceof AzureRepositoryNode) {
+            itemType = 'repository';
+        } else if (context && context instanceof AzureRepositoryNode) {
+            itemType = 'image';
+        } else {
+            itemType = 'registry';
+        }
+        vscode.window.showErrorMessage(`This ${itemType} has no associated build logs`);
+        return;
+    }
+
+    let links: { url?: string, id: number }[] = [];
+
+    links.sort((a, b): number => { return a.id - b.id });
+    let webViewTitle: string = registry.name;
+    if (context instanceof AzureRepositoryNode || context instanceof AzureImageNode) {
+        webViewTitle += (context ? '/' + context.label : '');
+    }
+    createWebview(webViewTitle, logData);
+
+}
+
+//# WEBVIEW COMPONENTS
+/** Generate the webview to display the logs */
+function createWebview(webviewName: string, logData: LogData): void {
```

Comment:
When the user clicks "Open Logs" within the Output Images box within the Build Tasks webview,
the new webview that pops up should be titled "myBuildName Logs"